packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -389,17 +422,22 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
 
   @override
   void dropChild(AbstractNode child) {
+    assert(!_debugMutationsLocked);
     if (!alwaysNeedsAddToScene) {
       markNeedsAddToScene();
     }
+
+    _incrementSubtreeCompositionObserverCount(-(child as Layer)._childrenWithCompositionCallbacks);
```

Comment:
We can probably type the "child" argument as Layer in the method signature, right?

Reply:
I think that will cause breakage in implementations that override these methods today, but it's probably worth it...